server/src/main/java/org/elasticsearch/action/bulk/BulkProcessor2.java

```diff
@@ -254,6 +254,17 @@ public void awaitClose(long timeout, TimeUnit unit) throws InterruptedException
         }
     }
 
+    /**
+     * This method flushes any requests, prevents any future retries, closes this processor, and returns once any pending requests complete.
+     */
+    public void close() {
+        try {
+            awaitClose(Integer.MAX_VALUE, TimeUnit.MILLISECONDS);
```

Comment:
Oh yeah also I think this should not wait so long especially since it's only used in tests. A minute should be more than enough. It would be good to return something from `awaitClose()` to indicate whether the wait was successful or not, so that you can e.g. `assertTrue(processor.awaitClose(1, TimeUnit.MINUTES))`. Also not sure this needs doing here, it seems independent of the other changes?

Reply:
I've actually got several related PRs going on at once. One of them (#94133, just merged to main) changes awaitClose to return a boolean. Another one (#94197) depends on having this close method. The DeprecationIndexingComponent actually calls close(), which is why I made the change here (and modified a BulkProcessor test just to make sure it had a little coverage). I've been going back and forth over what the correct behavior here ought to be -- wait forever? wait some amount of time? remove this method and make the client think about what is right for them (although that complicates the dboule try-with-resources in #94197)?

Reply:
It's usual to wait a bit (~30s say) but then to proceed regardless. We don't want to block the node shutting down forever, but bailing out straight away will sometimes drop some messages even if everything is working properly.